torchserve_dashboard/api.py

```diff
@@ -1,59 +1,73 @@
 
 import logging
 
-ENVIRON_WHITELIST = ["LD_LIBRARY_PATH", "LC_CTYPE", "LC_ALL", "PATH", "JAVA_HOME", "PYTHONPATH", "TS_CONFIG_FILE", "LOG_LOCATION", "METRICS_LOCATION"]
+ENVIRON_WHITELIST = [
+    "LD_LIBRARY_PATH", "LC_CTYPE", "LC_ALL", "PATH", "JAVA_HOME", "PYTHONPATH",
+    "TS_CONFIG_FILE", "LOG_LOCATION", "METRICS_LOCATION"
+]
 
 log = logging.getLogger(__name__)
 
 
 class LocalTS:
-    def __init__(self, model_store, config_path, log_location=None, metrics_location=None):
-        new_env = {}
+    def __init__(self,
+                 model_store: str,
+                 config_path: str,
+                 log_location: Optional[str] = None,
+                 metrics_location: Optional[str] = None) -> None:
+        new_env: Dict[str, Any] = {}
         env = os.environ
         for x in ENVIRON_WHITELIST:
             if x in env:
                 new_env[x] = env[x]
         if log_location:
             new_env["LOG_LOCATION"] = log_location
+            if not os.path.isdir(log_location):
+                os.makedirs(log_location, exist_ok=True)
         if metrics_location:
             new_env["METRICS_LOCATION"] = metrics_location
-        if not os.path.isdir(metrics_location):
-            os.makedirs(metrics_location, exist_ok=True)
-        if not os.path.isdir(log_location):
-            os.makedirs(log_location, exist_ok=True)
+            if not os.path.isdir(metrics_location):
+                os.makedirs(metrics_location, exist_ok=True)
 
         self.model_store = model_store
         self.config_path = config_path
         self.log_location = log_location
         self.metrics_location = metrics_location
         self.env = new_env
-    
-    def check_version(self):
+
+    def check_version(self) -> Tuple[str, Union[str, Exception]]:
         try:
-            p=subprocess.run(["torchserve","--version"], check=True,
-                            stdout=subprocess.PIPE,stderr=subprocess.PIPE,
-                            universal_newlines=True)
-            return p.stdout ,p.stderr
-        except (subprocess.CalledProcessError,OSError) as e:
-            return "",e
-         
-    def start_torchserve(self):
+            p = subprocess.run(["torchserve", "--version"],
+                               check=True,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               universal_newlines=True)
+            return p.stdout, p.stderr
+        except (subprocess.CalledProcessError, OSError) as e:
+            return "", e
+
+    def start_torchserve(self) -> str:
 
         if not os.path.exists(self.model_store):
             return "Can't find model store path"
         elif not os.path.exists(self.config_path):
             return "Can't find configuration path"
-        dashboard_log_path = os.path.join(self.log_location, "torchserve_dashboard.log")
+        dashboard_log_path: Optional[str] = os.path.join(
```

Comment:
Although I like type annotations for function input,outputs I find it slightly unnecessary to have typed inline(?) variables, at least for a project this size. 
IMO less is more at this point (I might be wrong)

Reply:
I personally don't have a problem with inline type annotations. This is mainly due to the used libraries not providing types.
But I have to admit that I mainly use inline type annotations for the VSCode type checker to be happy ^^.
We can skip these if you want. Feel free to remove them and push the changes :)